### DISCUSSION TITLE
unable to package(msix) winui3 c# project with winui3 c# class library project from CLI

### DISCUSSION THREAD

**Discussion by vash72** [microsoft/WindowsAppSDK#3749] [Q&A]
before adding the winui3 c# class library, i was able to produce the msix using the cli:

`dotnet msbuild <mainapp>.csproj /p:Configuration=Release /p:AppxPackageDir=.\dist\ /p:GenerateAppxPackageOnBuild=true /p:AppxBundlePlatforms=x64 /p:UapAppxPackageBuildMode=SideloadOnly /p:AppxBundle=Never /p:PackageCertificateKeyFile=<mainapp>_TemporaryKey.pfx /p:AppxPackageSigningEnabled=true`

i added a winui3 c# class library project to my solution, than i added a reference to ma winui3  c# class library project from the main app, and now i'm not able to build the project with cli:

the error from the command:
`error : You must include a valid app package manifest file named AppxManifest.xml in the source [library.csproj]`

may someone help me out to identify what i need to do in order to be able to properly build the misx with the library?

mainapp csproj:
```
	<PropertyGroup>
		<OutputType>WinExe</OutputType>
		<TargetFramework>net7.0-windows10.0.18362.0</TargetFramework>
		<TargetPlatformMinVersion>10.0.18362.0</TargetPlatformMinVersion>
		<RootNamespace>...removed...</RootNamespace>
		<ApplicationManifest>app.manifest</ApplicationManifest>
		<Platforms>x86;x64;</Platforms>
		<RuntimeIdentifiers>win10-x86;win10-x64;</RuntimeIdentifiers>
		<PublishProfile>win10-$(Platform).pubxml</PublishProfile>
		<UseWinUI>true</UseWinUI>
		<EnableMsixTooling>true</EnableMsixTooling>
		<Nullable>enable</Nullable>
		<PlatformTarget>x64</PlatformTarget>
		<RuntimeIdentifier>win10-x64</RuntimeIdentifier>
                <GenerateAppInstallerFile>False</GenerateAppInstallerFile>
                <AppxAutoIncrementPackageRevision>False</AppxAutoIncrementPackageRevision>
		<HoursBetweenUpdateChecks>0</HoursBetweenUpdateChecks>
		<AppxPackageSigningTimestampDigestAlgorithm>SHA256</AppxPackageSigningTimestampDigestAlgorithm>
		<AppxSymbolPackageEnabled>False</AppxSymbolPackageEnabled>
		<AppxBundle>Never</AppxBundle>
		<GenerateTestArtifacts>True</GenerateTestArtifacts>
	</PropertyGroup>
```

library csproj:
```
  <PropertyGroup>
    <TargetFramework>net7.0-windows10.0.18362.0</TargetFramework>
    <TargetPlatformMinVersion>10.0.18362.0</TargetPlatformMinVersion>
    <RootNamespace>...removed...</RootNamespace>
    <Nullable>enable</Nullable>
    <UseWinUI>true</UseWinUI>
    <Platforms>x86;x64;</Platforms>
    <RuntimeIdentifiers>win10-x86;win10-x64;</RuntimeIdentifiers>
    <PlatformTarget>x64</PlatformTarget>
    <RuntimeIdentifier>win10-x64</RuntimeIdentifier>
    <GenerateLibraryLayout>true</GenerateLibraryLayout>
    <EnableMsixTooling>true</EnableMsixTooling>
  </PropertyGroup>
```

**Comment by vash72** [comment]
the work around i found is to remove
    <EnableMsixTooling>true</EnableMsixTooling>
from the library csproj and add
    <EnableCoreMrtTooling Condition=" '$(BuildingInsideVisualStudio)' != 'true' ">false</EnableCoreMrtTooling>

**Reply by LucaZiegler** [reply]
Do you mean UseWinUI ?
Have the same issue with .NET MAUI...
